stdlib/2and3/codecs.pyi

```diff
@@ -192,13 +197,25 @@ class StreamReader(Codec):
     def reset(self) -> None:
         ...
 
-class StreamReaderWriter:
-    def __init__(self, stream: BinaryIO, Reader: _stream_reader_type, Writer: _stream_writer_type, errors: str = ...) -> None:
-        ...
-    def __enter__(self) -> BinaryIO:
-        ...
-    def __exit__(self, typ, exc, tb) -> bool:
-        ...
+_T = TypeVar('_T', bound='StreamReaderWriter')
+
+# Doesn't actually inherit from TextIO, but wraps a BinaryIO to provide text reading and writing
+# and delegates attributes to the underlying binary stream with __getattr__.
+class StreamReaderWriter(TextIO):
+    def __init__(self, stream: BinaryIO, Reader: _stream_reader_type, Writer: _stream_writer_type, errors: str = ...) -> None: ...
```

Comment:
Do you really need stream to be a `BinaryIO`? Or would `IO[bytes]` be enough?

(We have some Python 2 classes that only inherit from the latter, like tempfile._TemporaryFileWrapper, or cStringIO.InputType. Instances of those would not work as argument to StreamReaderWriter.__init__ with `stream` a `BinaryIO`.)